crm/crm/doctype/customer_feedback/customer_feedback.py

```diff
@@ -12,11 +12,20 @@
 
 
 class CustomerFeedback(Document):
+	selling_or_buying = "selling"
+
 	def validate(self):
 		self.set_missing_values()
 		self.set_title()
 		self.set_status()
 		self.get_previous_values()
+		self.set_reference_from_project()
+		if self.project and not (self.reference_doctype and self.reference_name):
+			self.set_reference_from_project()
```

Comment:
move condition inside method, only check if reference_name is empty, and call it before `set_missing_values` and project field does not exist in crm app, move to erpnext app